File: moviejava/app/src/main/java/com/example/moviejava/ui/home/adapter/PagerAdapter.java
package com.example.moviejava.ui.home.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.moviejava.R;
import com.example.moviejava.ui.home.fragment.genres.GenresFragment;
import com.example.moviejava.ui.home.fragment.now_playing.NowPlayingFragment;
import com.example.moviejava.ui.home.fragment.popular.PopularFragment;
import com.example.moviejava.ui.home.fragment.top_rate.TopRateFragment;
import com.example.moviejava.ui.home.fragment.upcoming.UpcomingFragment;

import javax.inject.Inject;


public class PagerAdapter extends
        FragmentStateAdapter {
    @Inject
    public PagerAdapter(FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 1:
                return new NowPlayingFragment();
            case 2:
                return new UpcomingFragment();
            case 3:
                return new TopRateFragment();
            case 4:
                return new GenresFragment();
            default:
                return new PopularFragment();
        }
    }


    @Override
    public int getItemCount() {
        return 5;
    }
}

File: movie/app/src/main/java/com/example/movie/ui/home/adapter/PagerAdapter.kt
package com.example.movie.ui.home.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.movie.ui.home.fragment.genres.GenresFragment
import com.example.movie.ui.home.fragment.now_playing.NowPlayingFragment
import com.example.movie.ui.home.fragment.popular.PopularFragment
import com.example.movie.ui.home.fragment.top_rate.TopRateFragment
import com.example.movie.ui.home.fragment.upcoming.UpcomingFragment
import javax.inject.Inject

class PagerAdapter @Inject constructor(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {

    override fun getItemCount(): Int = 5

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> PopularFragment()
            1 -> NowPlayingFragment()
            2 -> UpcomingFragment()
            3 -> TopRateFragment()
            4 -> GenresFragment()
            else -> PopularFragment()
        }
    }


}
